x/ccv/provider/keeper/grpc_query.go

```diff
@@ -182,3 +182,26 @@ func (k Keeper) QueryProposedConsumerChainIDs(goCtx context.Context, req *types.
 		ProposedChains: chains,
 	}, nil
 }
+
+func (k Keeper) QueryAllPairsValConAddrByConsumerChainID(goCtx context.Context, req *types.QueryAllPairsValConAddrByConsumerChainIDRequest) (*types.QueryAllPairsValConAddrByConsumerChainIDResponse, error) {
```

Comment:
Please try adding a basic unit test.

There are tests that show how to assign consumer keys. If you can, please confirm the behaviour with a testcase and then we're good to go.

Reply:
I see we lack a lot of tests for `queries`, we should make a pr to solve this @MSalopek

Reply:
Can you check this again @MSalopek